### PR TITLE
tests: fix test-awful-layout

### PR DESCRIPTION
**Pull request by blueyed**
TODO:
 - [ ] wait for investigation on https://github.com/awesomeWM/awesome/issues/1292.

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -222,9 +222,8 @@ for f in $tests; do
 
     if ! grep -q -E '^Test finished successfully$' "$awesome_log" ||
             grep -q -E '[Ee]rror|assertion failed' "$awesome_log"; then
-        echo "===> ERROR running $f! <==="
+        echo "===> ERROR running $f <==="
         grep --color -o --binary-files=text -E '.*[Ee]rror.*|.*assertion failed.*' "$awesome_log" || true
-        ((errors + 1))
```
Comment (psychon) [inline tests/run.sh]: What happened to this?

File: tests/run.sh [hunk 2]
```diff
@@ -224,6 +224,7 @@ for f in $tests; do
             grep -q -E '[Ee]rror|assertion failed' "$awesome_log"; then
         echo "===> ERROR running $f <==="
         grep --color -o --binary-files=text -E '.*[Ee]rror.*|.*assertion failed.*' "$awesome_log" || true
+        ((errors + 1))
```
Comment (psychon) [inline tests/run.sh]: A fixup gone wrong?

**Comment by psychon** [comment]
Besides the fixup gone wrong: :+1: 

I guess the issue is that clients are too slow at exiting (`c:kill()` takes some time to work) and we start running the next tests already before all clients are gone?

**Comment by blueyed** [comment]
There is no `c:kill()` involved here, is it?
It looks more like spawning 5 in the same loop does not work (for me).
Might be better after / fixed by https://github.com/awesomeWM/awesome/commit/fb874122ba1ac06a3af29c39c5abe6415e77bbf8.

**Comment by psychon** [comment]
The previous test might have opened clients. The test runner `c:kill()`s them after the test is done. However, I think that nothing makes sure the clients really close before the next test is started, hence the `if` in this test sees `#client.get() ~= 0` and doesn't start the `xterm`s.

At least that's my theory.

**Pull request closed without merging**